crates/turborepo-scm/src/package_deps.rs

```diff
@@ -31,19 +33,33 @@ impl SCM {
         inputs: &[S],
         telemetry: Option<PackageTaskEventBuilder>,
     ) -> Result<GitHashes, Error> {
+        // If the inputs contain "$TURBO_DEFAULT", we need to include the "default" file
+        // hashes as well. NOTE: we intentionally don't remove "$TURBO_DEFAULT"
+        // from the inputs if it exists in the off chance that the user has a
+        // file named "$TURBO_DEFAULT" in their package (pls no).
```

Comment:
If you don't remove it, but still set `include_default_files`, it'll _both_ include default files and pick up the file with that literal name? Could we add a test case for this crazy scenario?